aspnet/mvc/views/working-with-forms.rst

```diff
@@ -0,0 +1,486 @@
+
+
+.. contents:: Sections:
+  :local:
+  :depth: 1
+
+The Form Tag Helper
+---------------------
+  
+The `Form <https://www.w3.org/TR/html401/interact/forms.html>`__ Tag Helper:
+
+- Generates the HTML `<FORM> <https://www.w3.org/TR/html401/interact/forms.html>`__ ``action`` attribute value for a MVC controller action or named route
+- Generates a hidden `Request Verification Token <http://www.asp.net/mvc/overview/security/xsrfcsrf-prevention-in-aspnet-mvc-and-web-pages>`__ to prevent cross-site request forgery
+- Supports the Tag Helper attribute ``asp-route-returnurl`` which provides a return URL
+- HTML Helper alternative: ``Html.BeginForm``
+
+Sample:
+
+.. literalinclude::   forms/sample/final/Views/Demo/RegisterFormOnly.cshtml
+  :language: HTML
+
+The Form Tag Helper above generates the following HTML:
+ 
+.. code-block:: HTML
+
+  <form method="post" action="/Demo/Register">
+    <!-- Input and Submit elements -->
+    <input name="__RequestVerificationToken" type="hidden" value="<removed for brevity>" />
+   </form>
+  
+The MVC runtime generates the ``action`` attribute value from the Form Tag Helper attributes ``asp-controller`` and ``asp-action``. The Form Tag Helper also generates a hidden `Request Verification Token <http://www.asp.net/mvc/overview/security/xsrfcsrf-prevention-in-aspnet-mvc-and-web-pages>`__ to prevent cross-site request forgery. Protecting a pure HTML Form from cross-site request forgery is very difficult, the Form Tag Helper provides this service for you.
+
+Many of the views in the  *Views/Account* folder (generated when you create a new web app with *Individual User Accounts*)contain the ``asp-route-returnurl`` attribute:
+
+.. code-block:: HTML
+  :emphasize-lines: 2
+  
+  <form asp-controller="Account" asp-action="Login" 
+    asp-route-returnurl="@ViewData["ReturnUrl"]" 
+    method="post" class="form-horizontal" role="form">
+
+:Note: With the built in templates, ``returnUrl`` is only populated automatically when you try to access an authorized resource but are not authenticated or authorized. When you attempt an unauthorized access, the security middleware redirects you to the login page with the ``returnUrl`` set.
+
+Using a named route
+^^^^^^^^^^^^^^^^^^^
+
+The ``asp-route`` Tag Helper attribute can also generate markup for the HTML ``action`` attribute. An app with a :doc:`route </fundamentals/routing>`  named ``register`` could use the following markup for the registration page:
+ 
+.. literalinclude::  forms/sample/final/Views/Demo/RegisterRoute.cshtml 
+  :language: HTML
+  :emphasize-lines: 4
+
+The Input Tag Helper
+---------------------
+ 
+The Input Tag Helper binds an HTML `<input> <https://www.w3.org/wiki/HTML/Elements/input>`__ element to a model property in your razor view.
+
+The Input Tag Helper:
+
+- Generates the ``id`` and ``name`` HTML attributes for the model name specified in the ``asp-for`` attribute.
```

Comment:
@dougbu  how is this?
The data annotations applied to the `Email` and `Password` properties generate metadata on the model. The Input Tag Helper consumes the model metadata and produces HTML5 data-val-* attributes (see 🔧 Model Validation). (data-val- specifies data validation.) These attributes describe the validators to attach to the input fields. This provides unobtrusive HTML5 and jQuery validation. The unobtrusive attributes have the format data-val-rule="Error Message", where rule is the validation rule (such as required, email format, minimum string length, and maximum string length). If an error message is provided in the attribute, it is displayed as the value for the data-val-rule attribute.